Sources/SocksCore/TCPSocket.swift

```diff
@@ -42,8 +42,18 @@ extension TCPReadableSocket {
         let data = Bytes(capacity: maxBytes)
         let flags: Int32 = 0 //FIXME: allow setting flags with a Swift enum
         let receivedBytes = socket_recv(self.descriptor, data.rawBytes, data.capacity, flags)
-        guard receivedBytes > -1 else { throw SocksError(.readFailed) }
-        
+        guard receivedBytes != -1 else {
+            if errno == ECONNRESET {
+                // closed by peer, need to close this side. 
+                // Since this is not an error, no need to throw unless the close
```

Comment:
@cwoloszynski at the moment, this does not throw if close throws an error which conflicts with this comment, which should it be?